### PR TITLE
Add action buttons

### LINKED ISSUE #1018: Button for creating surveys, call assignments and tasks (old modal)
## Description
A user should be able to create surveys, call assignments and tasks on the "Activities" tab.

## Relevant Job Stories
* When I’m planning a campaign, I want to see where surveys fits/sits on the campaign timeline, so that I’ll be able to use surveys as parts of the bigger plan
* When I’m running a survey, I want to be able to keep it alongside related content such as call assignments and events, so that I can find them all in a single place

## Prerequisites
Don't start until #1015 has been merged.

## Requirements
- [x] The Campaign page header should contain a primary action button for creating a survey, call assignment or task.
- [x] The "Create Activity" button should open a drop-down where the user can select the type of activity to create
- [x] Creating a survey should create "optimistically" and navigate to the empty survey
- [x] Creating a call assignment should create "optimistically" and navigate to the empty assignment
- [x] Creating a task should (for now) open the existing create task modal

## Possible implementations
Reuse existing task modal for now.

## Design specifications

Image

## Open questions
TBD

### PR DESCRIPTION
**Pull request by niklasva82**
## Description
This PR adds an action button menu to create surveys, tasks, and call assignments.

## Screenshots
[Screencast from 2023-03-18 01-43-59.webm](https://user-images.githubusercontent.com/14962107/226074153-37663666-cc4b-4393-ad6e-e9ccc366c4b1.webm)

## Changes
* Adds new ZUIButtonMenu component
* Adds elements to that ButtonMenu to create surveys, call assignments and tasks
* Adds model, repo and store for campaign and uses it to create surveys and call assignments.
* Uses the old modal from the ZUISpeedDial to create tasks.
* Removes ZUISpeedDial from all single campaign view and campaigns calendar view which was missed before.

## Related issues
Resolves #1018

### CODE REVIEW

**Comment by niklasva82** [comment]
I noticed there was an issue with the menu being open after clicking, so I modified the ZUIButtonMenu to override the onClick prop of its children and close the menu before running the child's onClick. This works fine, but I have not been able to get Typescript to agree, I have no idea why it's complaining, but after trying a bunch of solutions, I ended up adding a //@ts-ignore

File: src/features/campaigns/store.ts [hunk 1]
```diff
@@ -32,5 +36,49 @@ const campaignsSlice = createSlice({
...
+    surveyCreate: (state) => {
+      state.surveyList.isLoading = true;
+    },
+    surveyCreated: (state, action: PayloadAction<ZetkinSurvey>) => {
+      const survey = action.payload;
+      state.surveyList.isLoading = false;
+      state.surveyList.items.push(remoteItem(survey.id, { data: survey }));
+    },
+  },
+});
```
Comment (richardolsson) [inline src/features/campaigns/store.ts:75]: What's the purpose of this? There is already a store slice for call assignments, and another one for surveys. I see no need for having a store slice called `campaign` where the same data is stored.

**Review by richardolsson — CHANGES_REQUESTED** [review]
I'm sorry to hear you struggled with typescript. Please reach out when you have problems like that. We really don't want to disable typescript anywhere, unless it's extremely special cases, and I don't think this is a valid such case.

I committed a change to that part of the code, which uses the same pattern as the similar `ZUIEllipsisMenu` component, in using a strongly typed `items` property instead of the vaguely typed children, to define the contents of the menu.

I also found something to be unexpected in your PR, which is that you've created a new store slice. Can you explain?

**Review by richardolsson — APPROVED** [review]
I've changed the things I did not like, so IMO it's ready for merging!

**Pull request merged**